### PR TITLE
[13.0][FIX] account_payment_order_notification: Create notification with the correct partner (invoice partner).

### PR DESCRIPTION
**Pull request by victoralmau**
Create notification with the correct partner (invoice partner).

Please @pedrobaeza and @CarlosRoca13 can you review it?

@Tecnativa TT40516

### CODE REVIEW

File: account_payment_order_notification/wizard/wizard_account_payment_order_notification.py [hunk 1]
```diff
@@ -30,7 +30,11 @@ def default_get(self, fields):
         )
         if po:
             line_ids = []
-            for partner in po.payment_line_ids.mapped("partner_id"):
+            # We need to transfer invoice partner so that the email is sent to the
+            # correct partner (not parent)
+            for partner in po.payment_line_ids.mapped(
+                "move_line_id.move_id.partner_id"
```
Comment (pedrobaeza) [inline account_payment_order_notification/wizard/wizard_account_payment_order_notification.py]: This is not correct: The move line may not be an invoice and don't have `partner_id`. You have to select the filled one (through a lambda in the mapped).

Comment (victoralmau) [reply account_payment_order_notification/wizard/wizard_account_payment_order_notification.py]: Changes done (I believe both use cases are taken into account).

File: account_payment_order_notification/wizard/wizard_account_payment_order_notification.py [hunk 2]
```diff
@@ -59,7 +63,8 @@ def action_process(self):
         notifications = []
         for item in self.line_ids.filtered("to_send"):
             payment_line_ids = self.order_id.payment_line_ids.filtered(
-                lambda x: x.partner_id == item.partner_id
+                lambda x: (x.move_line_id.move_id.partner_id or x.partner_id)
+                == item.partner_id
             )
             data = {
                 "partner_id": item.partner_id.id,
```
Comment (pedrobaeza) [inline account_payment_order_notification/wizard/wizard_account_payment_order_notification.py:69]: This one should change as well, isn't it? Can't you avoid the `or` each time?

Comment (victoralmau) [reply account_payment_order_notification/wizard/wizard_account_payment_order_notification.py:69]: IMO previously is correct, although a bit confusing, it is changed.

Comment (pedrobaeza) [reply account_payment_order_notification/wizard/wizard_account_payment_order_notification.py:69]: But you are not getting here the invoice partner...

Comment (victoralmau) [reply account_payment_order_notification/wizard/wizard_account_payment_order_notification.py:69]: `line_ids` records already have the correct `partner_id` (previously set in `default_get()`)

Comment (pedrobaeza) [reply account_payment_order_notification/wizard/wizard_account_payment_order_notification.py:69]: OK, got it.

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Fast-tracking the fix:

/ocabot merge patch

**Pull request merged**